invesalius/data/trackers.py

```diff
@@ -155,73 +187,74 @@ def PolarisP4Tracker(tracker_id):
+    if dlg_port.ShowModal() == ID_OK:
+        Cal_optitrack, User_profile_optitrack = dlg_port.GetValue()
+        try:
+            import optitrack
+            trck_init = optitrack.optr()
 
-    # return tracker initialization variable and type of connection
+            if trck_init.Initialize(Cal_optitrack, User_profile_optitrack)==0:
+                trck_init.Run() #Runs once Run function, to update cameras.
+            else:
+                trck_init = None
+        except ImportError:
+            print('Error')
+    else:
+        print('#####')
     return trck_init, 'wrapper'
 
-
-def ClaronTracker(tracker_id):
-    import invesalius.constants as const
-    from invesalius import inv_paths
-
+def ElfinRobot(robot_IP):
     trck_init = None
     try:
-        import pyclaron
-
-        lib_mode = 'wrapper'
-        trck_init = pyclaron.pyclaron()
-        trck_init.CalibrationDir = inv_paths.MTC_CAL_DIR.encode(const.FS_ENCODE)
-        trck_init.MarkerDir = inv_paths.MTC_MAR_DIR.encode(const.FS_ENCODE)
-        trck_init.NumberFramesProcessed = 1
-        trck_init.FramesExtrapolated = 0
-        trck_init.PROBE_NAME = const.MTC_PROBE_NAME.encode(const.FS_ENCODE)
-        trck_init.REF_NAME = const.MTC_REF_NAME.encode(const.FS_ENCODE)
-        trck_init.OBJ_NAME = const.MTC_OBJ_NAME.encode(const.FS_ENCODE)
+        import invesalius.data.elfin as elfin
+        print("Trying to connect Robot via: ", robot_IP)
+        trck_init = elfin.elfin_server(robot_IP, const.ROBOT_ElFIN_PORT)
         trck_init.Initialize()
-
-        if trck_init.GetIdentifyingCamera():
-            trck_init.Run()
-            print("MicronTracker camera identified.")
-        else:
-            trck_init = None
-
-    except ImportError:
-        lib_mode = 'error'
-        print('The ClaronTracker library is not installed.')
-
-    return trck_init, lib_mode
-
-
-def PolhemusTracker(tracker_id):
-    try:
-        trck_init = PlhWrapperConnection(tracker_id)
         lib_mode = 'wrapper'
-        if not trck_init:
-            print('Could not connect with Polhemus wrapper, trying USB connection...')
-            trck_init = PlhUSBConnection(tracker_id)
-            lib_mode = 'usb'
-            if not trck_init:
-                print('Could not connect with Polhemus USB, trying serial connection...')
-                trck_init = PlhSerialConnection(tracker_id)
-                lib_mode = 'serial'
+        print('Connect to elfin robot tracking device.')
+
     except:
+        lib_mode = 'disconnect'
         trck_init = None
-        lib_mode = 'error'
-        print('Could not connect to Polhemus by any method.')
+        print('Could not connect to elfin robot tracker.')
 
+    # return tracker initialization variable and type of connection
     return trck_init, lib_mode
 
+def RobotTracker(tracker_id):
+    from wx import ID_OK
+    trck_init = None
+    trck_init_robot = None
+    tracker_id = None
+    dlg_device = dlg.SetTrackerDevice2Robot()

```

Comment:
Full name: `SetTrackerDeviceToRobot`. Also, do we really need these 4 nested ifs? Can't it be simplified?